src/katgpucbf/xbgpu/main.py

```diff
@@ -445,6 +446,10 @@ async def async_main(args: argparse.Namespace) -> None:
 
         add_signal_handlers(xbengine)
         add_gc_stats()
+        # katsdpcontroller launches us with real-time scheduling, but we don't
+        # want that for the main Python thread since it can starve the
+        # latency-sensitive network threads.
+        os.sched_setscheduler(0, os.SCHED_OTHER, os.sched_param(0))
```

Comment:
Is this before or after the other threads are started?

Reply:
It's after the networking threads, which are created during the XBEngine constructor (either directly or in the BPipeline/XPipeline constructors).

There are a bunch of other helper threads that get created by CUDA, numpy and asyncio, and it's less clear whether they'll still have real-time scheduling priority. If I had unbounded time, I'd probably re-architect things so that we start with normal scheduling priority but with CAP_SYS_NICE, and then explicitly give real-time priority to the networking threads; but that would require modifications to at least katsdpcontroller, katgpucbf and spead2. Pragmatically, this is working for now.

Reply:
I approve of pragmatism.